cpu/rp2350/periph/gpio.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Tom Hert <git@annsann.eu>
+ * SPDX-FileCopyrightText: 2025 HAW Hamburg
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+#include "periph/gpio.h"
+
+#include <errno.h>
+
+#include "board.h"
+#include "irq.h"
+#include "periph_conf.h"
+#include "periph_cpu.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+#define GPIO_PIN_NUMOF 30U
+
+int gpio_init(gpio_t pin, gpio_mode_t mode) {
+  /* Check if we exceed the maximum number of GPIO pins */
+  assert(pin < GPIO_PIN_NUMOF);
+
+  /* Clear the pin's output enable and output state */
+  SIO->GPIO_OE_CLR = 1LU << pin;
+  SIO->GPIO_OUT_CLR = 1LU << pin;
+
+  switch (mode) {
+    case GPIO_OUT:
+      *(uint32_t*)calculate_gpio_io_ctrl_register_addr(pin) =
+          FUNCTION_SELECT_SIO;
+
+      volatile uint32_t* pad_reg =
+          (uint32_t*)calculate_gpio_pad_register_addr(pin);
+
+      /** We clear all bits except the drive strength bit 
```

Comment:
```suggestion
      /* We clear all bits except the drive strength bit 
```
Not a doxygen comment.

Reply:
Also there is a trailing whitespace (or more than one).

Reply:
fixed